core/src/main/java/org/infinispan/conflict/impl/StateReceiverImpl.java

```diff
@@ -164,25 +162,20 @@ InboundTransferTask createTransferTask(int segmentId, Address source, CacheTopol
          // If an exception is thrown by any of the inboundTransferTasks, then remove all segment results and cancel all tasks
          allSegmentRequests.exceptionally(throwable -> {
             if (trace) log.tracef(throwable, "Exception when processing InboundTransferTask for cache %s", cacheName);
-            synchronized (StateReceiverImpl.this) {
```

Comment:
@danberindei I also removed the two `synchronized (StateReceiverImpl.this)` blocks as they are redundant now that `stop()`, `onDataRehash(..)` and `getAllReplicasForSegment(...)` are no longer synchronized.